### PR TITLE
commit composer.lock file. Closes #217

### LINKED ISSUE #217: Commit composer.lock
Apparently committing the `composer.lock` file helps with `composer install`. This suggestion comes from the [composer project lead](https://twitter.com/seldaek/status/578687656910983168). This was an issue posted before by @romqin, but we had closed it with saying we were going to refactor.

### PR DESCRIPTION
**Pull request by james2doyle**
See #217

### CODE REVIEW

**Comment by Schlaefer** [comment]
:+1:

**Comment by Stijn-Flipper** [comment]
In addition, you should add the following entry to `.gitattributes`:

    composer.lock export-ignore

because the lock includes `require-dev` dependencies and you don't want to install that if you are using the released version (i.e. without development files). People also tend to modify `composer.json` for their own web project that depends on Phile and thus creating their own `composer.lock` instead.

In short, the lock is good for development, but leave it out when releasing it.

**Comment by Schlaefer** [comment]
> because the lock includes require-dev dependencies and you don't want to install that if you are using the released version

Well, but you don't have to install them. Locking the version doesn't change anything?

> People also tend to modify composer.json for their own web project that depends on Phile and thus creating their own composer.lock instead

So it's updated instead of created after the `composer` run.

> In short, the lock is good for development, but leave it out when releasing it.

I'm not convinced with this conclusion, at least not for the reasons above.
